internal/cfgplugins/dut_initialize.go

```diff
@@ -182,17 +162,233 @@ hardware tcam
+         packet ipv6 forwarding routed
+      feature tunnel vxlan
+         sequence 50
+         key size limit 160
+         packet ipv4 vxlan eth ipv4 forwarding routed decap
+         packet ipv4 vxlan forwarding bridged decap
+      feature vrf selection
+         port qualifier size 8 bits
+      feature vrf selection extended
+	  !
+	system profile vrf-selection-with-ip6-sip
+`
+	TcamProfilePolicyForwarding = `
+    hardware tcam
+  	profile tcam-policy-forwarding
+      feature traffic-policy port ipv4
+         sequence 45
+         key size limit 160
+         key field dscp dst-ip-label ip-frag ip-fragment-offset ip-length ip-protocol l4-dst-port-label l4-src-port-label src-ip-label tcp-control ttl
+         action count drop redirect set-dscp set-tc
+         packet ipv4 forwarding routed
+      !
+      feature traffic-policy port ipv6
+         sequence 25
+         key size limit 160
+         key field dst-ipv6-label hop-limit ipv6-length ipv6-next-header ipv6-traffic-class l4-dst-port-label l4-src-port-label src-ipv6-label tcp-control
+         action count drop redirect set-dscp set-tc
+         packet ipv6 forwarding routed
+      !
+   system profile tcam-policy-forwarding
+    !
+    hardware counter feature gre tunnel interface out
+    !
+    hardware counter feature traffic-policy in
+    !
+    hardware counter feature traffic-policy out
+    !
+    hardware counter feature route ipv4
+    !
+    hardware counter feature nexthop
+    !
     `
+)
 
-	if dut.Vendor() != ondatra.ARISTA || strings.ToLower(dut.Model()) == "ceos" {
-		t.Logf("TCAM profile not supported on %s %s", dut.Name(), dut.Model())
-		return
+func buildCliSetRequest(config string) *gpb.SetRequest {
+	gpbSetRequest := &gpb.SetRequest{
+		Update: []*gpb.Update{
+			{
+				Path: &gpb.Path{
+					Origin: "cli",
+					Elem:   []*gpb.PathElem{},
+				},
+				Val: &gpb.TypedValue{
+					Value: &gpb.TypedValue_AsciiVal{
+						AsciiVal: config,
+					},
+				},
+			},
+		},
 	}
+	return gpbSetRequest
+}
 
-	gnmiClient := dut.RawAPIs().GNMI(t)
-	t.Logf("Push the Tcam profile:%s", dut.Vendor())
-	gpbSetRequest := buildCliSetRequest(tcamProfileConfig)
-	if _, err := gnmiClient.Set(context.Background(), gpbSetRequest); err != nil {
-		t.Fatalf("Failed to set TCAM profile from CLI: %v", err)
+func ConfigureTcamProfile(t *testing.T, dut *ondatra.DUTDevice, tcamProfileConfig string) {
+	switch dut.Vendor() {
+	case ondatra.ARISTA:
+		if strings.ToLower(dut.Model()) == "ceos" {
+			t.Fatalf("TCAM profile not supported on %s %s", dut.Name(), dut.Model())
+		}
+		gnmiClient := dut.RawAPIs().GNMI(t)
+		t.Logf("Push the Tcam profile:%s", dut.Vendor())
+		gpbSetRequest := buildCliSetRequest(tcamProfileConfig)
+		if _, err := gnmiClient.Set(context.Background(), gpbSetRequest); err != nil {
+			t.Fatalf("Failed to set TCAM profile from CLI: %v", err)
+		}
+	default:
+		t.Fatalf("TCAM profile not supported on %s %s", dut.Name(), dut.Model())
```

Comment:
I believe we should by default, return an empty configuration.  If a DUT doesn't need any special hardware configuration (the factory default hardware initialization parameters, if any, are sufficient), then the test should be allowed to continue.